vmdb/app/services/orchestration_template_dialog_service.rb

```diff
@@ -0,0 +1,140 @@
+class OrchestrationTemplateDialogService
+  def create_dialog(dialog_label, template)
+    dialog = Dialog.new(:label => dialog_label, :buttons => "submit,cancel")
+    tab = dialog.dialog_tabs.build(:display => "edit", :label => "Basic Information", :position => 0)
+    add_stack_group(tab, 0)
+
+    template.parameter_groups.each_with_index do |parameter_group, index|
+      add_parameter_group(parameter_group, tab, index + 1)
+    end
+    dialog.save!
+    dialog
```

Comment:
I did not see this on the previous pass, but you could use the tap method here and put all the logic inside a block.  The tap method passes its object into the given block and, after the block finishes, returns the object.  Not required, but I like the look of the code this way.  What do you think?

Something like this:
```ruby
Dialog.new(:label => dialog_label, :buttons => "submit,cancel").tap do |dialog|
  tab = dialog.dialog_tabs.build(:display => "edit", :label => "Basic Information", :position => 0)
  add_stack_group(tab, 0)
  template.parameter_groups.each_with_index do |parameter_group, index|
    add_parameter_group(parameter_group, tab, index + 1)
  end
  dialog.save!
end
```